### DISCUSSION TITLE
I made a docker image for this

### DISCUSSION THREAD

**Discussion by GaryLee1101** [MoatLab/FEMU#55] [General]
This is an exciting job, but it's not so easy to install it. So I made a docker image for it, already uploaded to docker hub. I made it based on a Python image, so the os is Debian.

You can get it by:
 `docker pull lijiali1101/femu`
Besides, you should give the device folder permission to it, you may run it by this command:
`docker run --privileged -it -v /dev:/dev lijiali1101/femu /bin/bash`

The code is stored in /home/Code/femu, and I have modified the shell file, remove the "sudo" in each script. So you can exec "./run-blackbox.sh" and the other two programs directly.

I use the qcow2 file provided by the author, so the user/password both are "femu".

I have tested it several times, it worked fine. Hope the installation will no longer trouble ^_^

**Comment by huaicheng** [comment]
Sounds great, thanks!

I will keep this issue open in case others are interested.

**Comment by wczzsdsj** [comment]
Hello, I have a question. Can I use the femu script in this image to simulate ZNS? (I started the image, but so far I only see other scripts and not run-zns.sh)

**Reply by GaryLee1101** [reply]
I built this image years ago, the FEMU repository have added many features, including ZNS. You can try to pull the FEMU repository in this image to run, but  I don't know if the reliance have changed. So you'd better build a new image and install FEMU in it, it should be a more efficient way. Install FEMU in docker is simple.
